vllm/model_executor/layers/quantization/fp8.py

```diff
@@ -137,10 +137,35 @@ def from_config(cls, config: dict[str, Any]) -> "Fp8Config":
                    ignored_layers=ignored_layers,
                    weight_block_size=weight_block_size)
 
+    def get_xpu_quant_method(self, layer: torch.nn.Module,
+                             prefix: str) -> Optional["QuantizeMethodBase"]:
+        from vllm.attention.layer import Attention
+        from vllm.model_executor.layers.quantization.ipex_quant import (
+            IPEXFp8LinearMethod, IPEXFp8MoEMethod)
+        fp8_config = Fp8Config(
+            is_checkpoint_fp8_serialized=self.is_checkpoint_fp8_serialized,
+            activation_scheme=self.activation_scheme,
+            ignored_layers=self.ignored_layers,
+            weight_block_size=self.weight_block_size)
+
+        if isinstance(layer, LinearBase):
+            if is_layer_skipped(prefix=prefix,
+                                ignored_layers=self.ignored_layers,
+                                fused_mapping=self.packed_modules_mapping):
+                return UnquantizedLinearMethod()
+            return IPEXFp8LinearMethod(fp8_config)
+        elif isinstance(layer, FusedMoE):
+            return IPEXFp8MoEMethod(fp8_config, layer)
+        elif isinstance(layer, Attention):
+            return Fp8KVCacheMethod(self)
+        return None
+
     def get_quant_method(self, layer: torch.nn.Module,
                          prefix: str) -> Optional["QuantizeMethodBase"]:
         from vllm.attention.layer import Attention  # Avoid circular import
 
+        if current_platform.is_xpu():
+            return self.get_xpu_quant_method(layer, prefix)
```

Comment:
@ProExpertProg @bnellnm @LucasWilkinson Thanks for your review. I did a further refactor for xpu to avoid too many dispatches in current fp8 path. But since not like quantized models, there is no override mechanism for fp8 online quant so I still need such a router to xpu like here. Please take a review again.